src/main/kotlin/ecommerce/api/ProductService.kt

```diff
@@ -0,0 +1,67 @@
+package ecommerce.api
+
+import org.springframework.jdbc.core.JdbcTemplate
+import org.springframework.jdbc.core.RowMapper
+import org.springframework.stereotype.Service
+import java.sql.ResultSet
+
+@Service
+class ProductService(private val db: JdbcTemplate) {
+    private val productRowMapper =
+        RowMapper<Product> { rs: ResultSet, _ ->
+            Product(
+                rs.getLong("id"),
+                rs.getString("name"),
+                rs.getDouble("price"),
+                rs.getString("imageUrl"),
+            )
+        }
+
+    fun findAll(): List<Product> {
+        val products = db.query("""select * from product;""", productRowMapper)
+        return products
+    }
+
+    fun findById(id: Long): Product? {
+        var product: Product?
+        try {
+            product =
+                db.queryForObject(
+                    """select id, name, price, imageUrl from product where id = ?""",
+                    productRowMapper,
+                    id,
+                )
+        } catch (exception: Exception) {
+            println(exception.message)
+            return null
+        }
+        return product
+    }
```

Comment:
I think it'll be simpler for us to return the result right away.
```suggestion
    fun findById(id: Long): Product? {
        try {
            return db.queryForObject(
                    """select id, name, price, imageUrl from product where id = ?""",
                    productRowMapper,
                    id,
                )
        } catch (exception: Exception) {
            println(exception.message)
            return null
        }
    }
```

Reply:
Yes, it is better.